### PR TITLE
field don't show dict_id, dict_is_ordered in `format/debug`

### PR DESCRIPTION
**Pull request by jackwener**
# Which issue does this PR close?

Closes #.

# Rationale for this change

# What changes are included in this PR?

field don't show dict_id, dict_is_ordered in format/debug

Make field format clear

# Are there any user-facing changes?

Yes, it will change output.

### CODE REVIEW

File: arrow-schema/src/field.rs [hunk 1]
```diff
@@ -543,10 +543,29 @@ impl Field {
     }
 }
 
-// TODO: improve display with crate https://crates.io/crates/derive_more ?
+impl std::fmt::Debug for Field {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match &self.data_type {
+            DataType::Dictionary(_, _) => write!(f, "{self:?}"),
```
Comment (tustvold) [inline arrow-schema/src/field.rs:549]: As pointed out by clippy this will recurse infinitely

File: arrow-schema/src/field.rs [hunk 2]
```diff
@@ -543,10 +543,23 @@ impl Field {
...
+            "Field {{ name: \"{}\", data_type: {:?}, nullable: {}, metadata: {:?} }}",
+            self.name, self.data_type, self.nullable, self.metadata
+        )
+    }
+}
+
 impl std::fmt::Display for Field {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        write!(f, "{self:?}")
+        write!(
```
Comment (tustvold) [inline arrow-schema/src/field.rs:558]: I think this could just use the debug impl?

File: arrow-schema/src/field.rs [hunk 3]
```diff
@@ -543,10 +543,23 @@ impl Field {
     }
 }
 
-// TODO: improve display with crate https://crates.io/crates/derive_more ?
+impl std::fmt::Debug for Field {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(
```
Comment (tustvold) [inline arrow-schema/src/field.rs:548]: I think we should still print the dict id and ordered state if the datatype is a dictionary

**Comment by jackwener** [comment]
After careful consideration, I believe that this PR would result in many changes, but the benefits are not as substantial. Therefore, I have decided to close it.

**Pull request closed without merging**